lib/manageiq/messaging/kafka/common.rb

```diff
@@ -27,47 +32,55 @@ def queue_consumer
         end
 
         def raw_publish(commit, body, options)
-          producer.produce(encode_body(body), options)
+          producer.produce(encode_body(options[:headers], body), options)
           producer.deliver_messages if commit
-          logger.info("Published to topic(#{options[:topic]}), msg(#{payload_log(body[:data].inspect)})")
+          logger.info("Published to topic(#{options[:topic]}), msg(#{payload_log(body.inspect)})")
         end
 
         def queue_for_publish(options)
           body, kafka_opts = for_publish(options)
-          body[:message_type] = options[:message] if options[:message]
-          body[:class_name] = options[:class_name] if options[:class_name]
+          kafka_opts[:headers] = Hash(kafka_opts[:headers]).merge(:message_type => options[:message]) if options[:message]
```

Comment:
Can `kafka_opts[:headers]` be set before this?  Could you do
```
kafka_opts[:headers] = {}
kafka_opts[:headers].merge(:message_type => ...) if ...
kafka_opts[:headers].merge(:class_name => ...) if ...
```

Doing `Hash(kafka_opts[:headers]).merge` relying on the fact that it will be `Hash(nil)` looks more complicated than it is worth.